### PR TITLE
Add fax support, improve type checking, and enable Ray for background tasks

### PR DESCRIPTION
**Pull request by holdenk**
Add support for sending faxes & improve the type checking we do (check untyped functions) after spending some time dealing with something that mypy would have caught there.

This also adds a Ray cluster to the deployment for background tasks. Mostly because celery was annoying.

### CODE REVIEW

**Comment by holdenk** [comment]
The next parts to add:
1) Click button to include request for claim file in the appeal request
2) Earlier button for (toll-free numbers only) to request the claim file _before_ filing the appeal

File: fighthealthinsurance/ml_models.py [hunk 1]
```diff
@@ -216,6 +216,13 @@ def _clean_diagnosis_response(self, response):
             return None
         return self.diagnosis_response_regex.sub("", response)
 
+    @cache
+    def get_fax_number(self, denial: str) -> Optional[str]:
+        return self._infer(
+            system_prompt="You are a helpful assistant.",
+            prompt=f"Tell me what the to appeal fax number in the provided denial. If unknown write UNKNOWN. If known just without any pre-amble and as a snippet from the original doc. The denial follows: {denial}",
```
Comment (marcus-campbell) [inline fighthealthinsurance/ml_models.py:223]: We'll probably want to clean up the grammar here.

Should be: "Tell me the to appeal fax number _is_ within the provided denial. If the fax number is unknown write UNKNOWN. If known just output the fax number without any pre-amble and as a snipper from the original doc..."

File: fighthealthinsurance/templates/faxfollowup.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{% endblock title %}
+
+{% load compress %}
+{% load static %}
+
+{% block scriptmagic %}
+{% endblock scriptmagic %}
+{% block content %}
+<h2>Resend Your Fax</h2>
+<p>Thank you so much for us fight health insurance.
```
Comment (marcus-campbell) [inline fighthealthinsurance/templates/faxfollowup.html]: Looks like a typo?

File: fighthealthinsurance/ml_models.py (hunk 1 above)
Comment (marcus-campbell) [inline fighthealthinsurance/ml_models.py]: We'll probably want to clean up the grammar here.

Should be: "Tell me the to appeal fax number is within the provided denial. If the fax number is unknown write UNKNOWN. If known just output the fax number without any pre-amble and as a snipper from the original doc..."

File: build_ray.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -ex
+
+# Ray doesn't publish combiend images because idk.
+RAY_VERSION=2.38.0-py311
+RAY_IMAGE=holdenk/ray:${RAY_VERSION}
+RAY_IMAGE=holdenk/ray:${RAY_VERSION}
```
Comment (marcus-campbell) [inline build_ray.sh]: `RAY_IMAGE` is assigned twice

File: fighthealthinsurance/email_polling_actor_ref.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    @cached_property
+    def get(self):
+        # Shut down existing actor
+        try:
+            a = ray.get_actor(name, namespace="fhi")
+            if a is not None:
+                ray.kill(a)
+                time.sleep(10)
+        except Exception as e:
+            print(f"No exisitng email actor to stop {e}")
```
Comment (marcus-campbell) [inline fighthealthinsurance/email_polling_actor_ref.py]: small typo in error message

File: fighthealthinsurance/fax_actor_ref.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+class FaxActorRef:
+    @cached_property
+    def get(self):
+
+        # Shut down existing actor if needed
+        name = "FaxActor"
+        fax_actor = None
+        try:
+            fax_actor = ray.get_actor(name, namespace="fhi")
```
Comment (marcus-campbell) [inline fighthealthinsurance/fax_actor_ref.py:16]: Don't think we've set RAY_GCS_SERVER_REQUEST_TIMEOUT_SECONDS anywhere, so `get_actor()` could potentially timeout at up to 60 seconds, right?

**Review by marcus-campbell — APPROVED** [review]
Awesome work. Curious about expanding testing but this is great

File: fighthealthinsurance/fax_actor_ref.py (hunk 5 above)
Comment (holdenk) [reply fighthealthinsurance/fax_actor_ref.py:16]: It's true, we might want to decrease that timeout.

**Comment by holdenk** [comment]
I've been thinking about (as a follow up) only displaying the fax option in the flow if the ray cluster is up (but I was also thinking of adding a polling worker similar to the email worker so it wouldn't matter).

File: build_ray.sh (hunk 3 above)
Comment (holdenk) [reply build_ray.sh]: Ah good catch :)

File: fighthealthinsurance/ml_models.py (hunk 1 above)
Comment (holdenk) [reply fighthealthinsurance/ml_models.py]: Good call. I think we can probably switch these to the templating subsystem at some point too to make it easier if that's something you want to do.

File: fighthealthinsurance/templates/faxfollowup.html (hunk 2 above)
Comment (holdenk) [reply fighthealthinsurance/templates/faxfollowup.html]: Yup should have been using

**Pull request merged**